tests/integration-tests/tests/update/test_update.py

```diff
@@ -439,3 +441,144 @@ def get_batch_spot_bid_percentage(stack_name, region):
+    )
+
+    # wait until job running
+    scheduler_commands.wait_job_running(new_ami_job)
+    # cancel job in queue1
+    scheduler_commands.cancel_job(queue1_job_id)
+
+    # check the new launching instances are using new amis
+    _check_queue_ami(cluster, ec2, pcluster_ami_id, "queue1")
+    _check_queue_ami(cluster, ec2, pcluster_copy_ami_id, "queue2")
+    assert_compute_node_states(scheduler_commands, queue1_nodes, "idle")
+
+    # assert queue drain strategy doesn't trigger protected mode
+    assert_no_msg_in_logs(
+        remote_command_executor,
+        ["/var/log/parallelcluster/clustermgtd"],
+        ["Node bootstrap error"],
+    )
+
+
+def _test_update_without_queue_strategy(pcluster_config_reader, pcluster_ami_id, pcluster_copy_ami_id, cluster):
+    """Test update without setting queue strategy, update will fail."""
+    updated_config_file = pcluster_config_reader(
+        config_file="pcluster.config.update.yaml", global_custom_ami=pcluster_ami_id, custom_ami=pcluster_copy_ami_id
+    )
+    response = cluster.update(str(updated_config_file), raise_on_error=False)
+    assert_that(response["message"]).is_equal_to("Update failure")
+    assert_that(response.get("updateValidationErrors")[0].get("message")).contains(
+        "All compute nodes must be stopped or QueueUpdateStrategy must be set"
+    )
+
+
+def _check_queue_ami(cluster, ec2, ami, queue_name):
+    """Check if the ami of the queue instances are expected"""
+    instances = cluster.get_cluster_instance_ids(node_type="Compute", queue_name=queue_name)
+    _check_instance_ami_id(ec2, instances, ami)
+
+
+def _test_queue_strategy_without_running_job(
+    pcluster_config_reader, pcluster_ami_id, cluster, region, os, remote_command_executor, scheduler_commands
+):
+    """Test queue parameter update with drain stragegy without running job in the queue."""
+    updated_config_file = pcluster_config_reader(
+        config_file="pcluster.update_drain_without_running_job.yaml",
+        global_custom_ami=pcluster_ami_id,
+    )
+    cluster.update(str(updated_config_file))
+    # check chef client log contains expected log
+    assert_errors_in_logs(
+        remote_command_executor,
+        ["/var/log/chef-client.log"],
+        ["Queue update strategy is \\(DRAIN\\)", "Adding queue \\(queue1\\) to list of queue to be updated"],
+    )
+    queue1_nodes = scheduler_commands.get_compute_nodes("queue1")
+    wait_for_compute_nodes_states(scheduler_commands, queue1_nodes, expected_states=["idle", "idle~"])
+
+    instances = cluster.get_cluster_instance_ids(node_type="Compute", queue_name="queue1")
+    for instance in instances:
+        root_volume_id = utils.get_root_volume_id(instance, region, os)
+        volume_size = (
+            boto3.client("ec2", region_name=region)
+            .describe_volumes(VolumeIds=[root_volume_id])
+            .get("Volumes")[0]
+            .get("Size")
+        )
+        assert_that(volume_size).is_equal_to(40)
```

Comment:
we need to be sure this value (40) is different from the default, e.g. if in future the default will be 40 and then configuration used for the update will have no changes. To solve this we can explicitly set the size for the initial config.
minor - let's put 40 in one common place in the code and pass the value to the conf, so that if it's going to be changed, we need to change it in one place.

Reply:
Done